### PR TITLE
Header as search parameter

### PR DESCRIPTION
**Pull request by pmaasz**
Adds Header as search parameter.

### CODE REVIEW

File: src/Search/Header/Header.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace Ddeboer\Imap\Search\Text;
+
+use Ddeboer\Imap\Search\AbstractText;
+
+/**
+ * Class Header
+ *
+ * @author Philip Maaß <PhilipMaaß@aol.com>
+ */
+class Header extends AbstractText
```
Comment (Slamdunk) [inline src/Search/Header/Header.php]:
```suggestion
final class Header extends AbstractText
```

File: src/Search/Header/Header.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+    /**
+     * @return string
+     */
+    protected function getKeyword(): string
+    {
+        return "HEADER";
```
Comment (Slamdunk) [inline src/Search/Header/Header.php]: Please run `vendor/bin/php-cs-fixer fix` as suggested by Travis

File: src/Search/Header/Header.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Ddeboer\Imap\Search\Text;
```
[the remark this answers is not in the document]
Comment (pmaasz) [reply src/Search/Header/Header.php]: yes I saw it just in this moment too. Thanks for the fast reply.

**Comment by Slamdunk** [comment]
It seems `HEADER` is not a standard search criterion, maybe it's affected by the same issue as `OR`: https://github.com/ddeboer/imap#unknown-search-criterion-or

**Comment by pmaasz** [comment]
Seems like it. What do we do now? The desired changes are available in my fork

**Comment by Slamdunk** [comment]
I cannot merge this PR as the feature isn't proved to be working, but you can always use your class in your code (with a different namespace of course) if it works on your custom IMAP built.

**Comment by pmaasz** [comment]
I understand. Thanks for your fast feedback.

**Pull request closed without merging**